tests/integration/targets/consul/tasks/consul_agent_service.yml

```diff
@@ -0,0 +1,176 @@
+---
+# Copyright (c) 2024, Michael Ilg (@Ilgmi)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Create a service with name nginx
+  community.general.consul_agent_service:
+    name: nginx
+    service_port: 80
+    address: localhost
+    tags:
+      - http
+    meta:
+      nginx_version: 1.25.3
+    checks:
+      - interval: 60s
+        notes: "My Note"
+        http: http://localhost:80/morestatus
+    token: "{{ consul_management_token }}"
+  register: result
+
+- set_fact:
+    nginx_service: "{{result.service}}"
+
+- assert:
+    that:
+      - result is changed
+      - result.service.ID is defined
+      - result.service.Service == 'nginx'
+      - result.service.Address == 'localhost'
+      - result.service.Port == 80
+      - result.service.Tags[0] == 'http'
+      - result.service.Meta.nginx_version is defined
+      - result.service.Meta.nginx_version == '1.25.3'
+      - result.service.ContentHash is defined
+
+- name: Update a service with name {{ nginx_service.Service }}
```

Comment:
We should also add idempotency checks here, ie ensure that running this task twice doesn't result in updates twice.

Reply:
I needed some help with the update method. 
The module object and the API object are not similar. For example, the name will become service. Another point is that there is no definition of the checks in the API object. So, we need to get all checks and check if they need an update. 
One easy solution is to change the check_mode to partial and constantly update the object.